tests/test_streaming_file_reader.py

```
@@ -0,0 +1,681 @@
+            shutil.rmtree(temp_dir)
+
+    @pytest.mark.asyncio
+    async def test_large_file_memory_bound(self):
+        """
+        Verifies that the streaming file reader can successfully process a very large file (>1MB) without errors, ensuring content completeness and expected line count.
+        """
+        temp_dir = tempfile.mkdtemp()
+
+        # Create a large file (simplified version - 2MB for faster testing)
+        large_file = Path(temp_dir) / "very_large.txt"
+        with open(large_file, "w") as f:
```

Pylint (3.3.7)

[warning] Using open without explicitly specifying an encoding

(W1514)